op-batcher/batcher/driver.go

```diff
@@ -527,7 +527,7 @@ func (l *BatchSubmitter) sendTransaction(ctx context.Context, txdata txData, que
 			ChannelId:       txdata.ID().chID,
 			FrameNumber:     txdata.ID().frameNumber,
 			FrameDataLength: uint32(txdata.Len()),
-			IsLast:          false,
+			IsLast:          txdata.isCloseFrame,
 			NumL2Blocks:     1,
```

Comment:
I was going to say we still need to collect this, but... is it always going to be possible? Do we actually collect multiple L2 blocks into one L1 transaction in practice?

Reply:
Currently tests fail without this, so I guess our version of the code does not ensure that channels are opened and closed in the same L1 tx?

Reply:
The guarantee is that they contain a single _batch_, but I thought that there was one L1 tx per batch. I was asking about line 531, not line 530. Failing because of 530 makes sense, because you might have multiple frames for a large batch.

Reply:
I see. Is this field NumL2Blocks related to this ticket? #38

Reply:
I believe so. If it is, the best solution might be to remove the field. There is no good way for the contract to collect this count, and we already determined that L1 blocks was the correct metric, not L2 blocks.

Reply:
Right, the field NumL2Blocks has been removed in 8d55bbf36.